entity-framework/core/what-is-new/ef-core-7.0/whatsnew.md

```diff
@@ -19,6 +19,316 @@ EF7 is also available as [daily builds](https://github.com/dotnet/efcore/blob/ma
+
+<!--
+        await context.Set<FeaturedPost>().ExecuteDeleteAsync();
+-->
+[!code-csharp[DeleteFeaturedPosts](../../../../samples/core/Miscellaneous/NewInEFCore7/ExecuteDeleteSample.cs?name=DeleteFeaturedPosts)]
+
+Generates the following SQL when using TPH mapping:
+
+```sql
+DELETE FROM [p]
+FROM [Posts] AS [p]
+WHERE [p].[Discriminator] = N'FeaturedPost'
+```
+
+There are also no issues for this case when using the TPC mapping strategy, since again only changes to a single table are needed:
+
+```sql
+DELETE FROM [f]
+FROM [FeaturedPosts] AS [f]
+```
+
+However, attempting this when using the TPT mapping strategy will fail since it would require deleting rows from two different tables.
+
+Adding a filter to the query often means the operation will fail with both the TPC and TPT strategies. This is again because the rows may need to be deleted from multiple tables. For example, this query:
+
+<!--
+        await context.Posts.Where(p => p.Author.Name.StartsWith("Arthur")).ExecuteDeleteAsync();
+-->
+[!code-csharp[DeletePostsForGivenAuthor](../../../../samples/core/Miscellaneous/NewInEFCore7/ExecuteDeleteSample.cs?name=DeletePostsForGivenAuthor)]
+
+Generates the following SQL when using TPH:
+
+```sql
+DELETE FROM [p]
+FROM [Posts] AS [p]
+LEFT JOIN [Authors] AS [a] ON [p].[AuthorId] = [a].[Id]
+WHERE [a].[Name] IS NOT NULL AND ([a].[Name] LIKE N'Arthur%')
+```
+
+But fails when using TPC or TPT.
+
+> [!TIP]
+> [Issue #10879](https://github.com/dotnet/efcore/issues/28520) tracks adding support for automatically sending multiple commands in these scenarios. Vote for this issue if it's something you would like to see implemented.
+
+### `ExecuteDelete` and relationships
+
+As mentioned above, it may be necessary to delete or update dependent entities before the principal of a relationship can be deleted. For example, each `Post` is a dependent of its associated `Author`. This means that an author cannot be deleted if a post still references it; doing so will violate the foreign key constraint in the database. For example, attempting this:
+
+```csharp
+await context.Authors.ExecuteDeleteAsync();
+```
+
+Will result in the following exception on SQL Server:
+
+> Microsoft.Data.SqlClient.SqlException (0x80131904): The DELETE statement conflicted with the REFERENCE constraint "FK_Posts_Authors_AuthorId". The conflict occurred in database "TphBlogsContext", table "dbo.Posts", column 'AuthorId'.
+The statement has been terminated.
+
+To fix this, we must first either delete the posts, or sever the relationship between each post and its author by setting `AuthorId` foreign key property to null. For example, using the delete option:
```

Comment:
I'd put the database cascading delete solution first, since it's really the thing to do wheter supported. Honestly, that's the universal relational way to do this, and this whole complexity is only because SQL Server doesn't support cycles... Like in PG I'd always recommend cascade deletes...

Reply:
I'd rather leave it the way it is, because talking about the problem first shows why cascades are useful.

Reply:
Makes sense!